addons/account/models/account_account.py

```diff
@@ -160,10 +160,10 @@ def _field_to_sql(self, alias: str, fname: str, query: (Query | None) = None, fl
                           FROM account_account_res_company_rel rel
                           JOIN res_company
                             ON res_company.id = rel.res_company_id
-                         WHERE rel.res_company_id IN %(authorized_company_ids)s
+                         WHERE rel.res_company_id = ANY(%(authorized_company_ids)s)
                       ORDER BY rel.account_account_id, company_id
                     )""",
-                    authorized_company_ids=self.env.user._get_company_ids(),
+                    authorized_company_ids=list(self.env.companies.ids),
```

Comment:
Can you explain why you want to change this?
The messages says
> sort is done via computed field `placeholder_code`, but what does that change?

I guess we need this as it is now to be able to search on the placeholder, and for the value not to change depending on the context.

@antoine162 can probably explain better than me

Reply:
@william-andre I was thinking it's okay if the sort order changes depending on the context :smile: 

The code and placeholder code are context-dependent, and the whole purpose of that sorting is to display the accounts in an order which is consistent with the codes in the CoA list view, so it's logical that the sorting would also be context-dependent.

@KangOl I was going to do something similar in this open PR:
https://github.com/odoo/odoo/blob/8c9a8a44a0ca3ec4c26fdcbfed7646c58759ec46/addons/account/models/account_account.py#L173

We just need to be careful about handling branches: you can consider the following case which I think my open PR will handle correctly:
- The user has access to 3 companies: Company 1, Company 2 and Company 2.1 which is a branch of Company 2
- Active companies are Company 1 (first active company) and Company 2.1
- Account A belongs to Company 2 and has code 123456 in Company 2. It's visible because Company 2.1 is active and because branches have access to their parent companies' accounts.
- We want the placeholder_code to be `123456 (Company 2.1)`.
- But if we restrict authorized_company_ids to just `self.env.companies.ids`, then the WHERE clause will filter out everything because none of the active companies are in the company_ids of the account.

My open PR solves this by considering all the parent companies of the active companies :smile: